### PR TITLE
doc, rpc: Release notes and follow-ups for #29612

### PR DESCRIPTION
**Pull request by fjahr**
This adds release notes for #29612 and addresses post-merge review comments.

### CODE REVIEW

**Review by maflcko — APPROVED** [review]
lgtm, apart from the assert

**Review by maflcko — APPROVED** [review]
utACK efc1b5be8a4696c0db19ba18316b2d4ed09e10f2

**Comment by fanquake** [comment]
@theStack or @TheCharlatan want to have a look over the followup?

**Review by theStack — APPROVED** [review]
utACK efc1b5be8a4696c0db19ba18316b2d4ed09e10f2

File: doc/release-notes-29612.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+RPC
+---
+
+- The `dumptxoutset` RPC now returns the UTXO set dump in a new and
+  improved format. At the same time the `loadtxoutset` RPC now
```
[the remark this answers is not in the document]
Comment (fanquake) [reply doc/release-notes-29612.md:5]: I think we can make any further release note additions when the notes are consolidated / merged into the wiki.

**Pull request merged**